doc/build-helpers/fetchers.chapter.md

```diff
@@ -17,28 +13,90 @@ The differences are explained in this list:
 | `pkgs.fetch*` | build time | derivation | `/nix/store`, substituters | output store object not available |
 
+:::{.note}
+`pkgs.fetchFrom*` helpers retrieve _snapshots_ of version-controlled sources, as opposed to the entire version history, which is more efficient.
+:::
+
 ## Caveats {#chap-pkgs-fetchers-caveats}
 
-Because Nixpkgs fetchers are [fixed-output derivations](https://nixos.org/manual/nix/stable/glossary#gloss-fixed-output-derivation), they need you to specify a hash (this is what turns them into a fixed-output derivation).
-This is usually done through a `hash` attribute, which **can be different from the hash of the contents that are downloaded**.
-On top of that, Nix checks if there is already an object in the Nix store with the same hash, and reuses that object if it exists.
+Because Nixpkgs fetchers are fixed-output derivations, an [output hash](https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-outputHash) has to be specified.
+This hash refers to the derivation output, not the remote source itself!
+
 This has the following implications that you should be aware of:
 
-- In most cases, it's better to use Nix (or Nix-aware) tooling to give you the hash, because you may not be able to just hash the contents and use that result.
-  See [the tip](#fetchers-hash-tip) below for an easy way to do this.
-  If you can't use the method suggested by the tip, you should understand the hash [algorithm](https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-outputHashAlgo) and [mode](https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-outputHashMode) used by the fetcher you're using so you can calculate the hash by yourself beforehand.
-- If you update the URL of the content to be fetched (or a version parameter, or some other parameter), you **must not** leave the hash unchanged, because it will reuse the old contents that were already fetched (this will lead to unexpected behaviour when building the derivation).
-  If the contents with the given hash already exist, Nix won't fetch any content again (even if the URL changed!) and will reuse the existing contents.
-  The Nix manual [explains](https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-outputHash) why Nix ignores URL changes in this case.
+- Use Nix (or Nix-aware) tooling to produce the output hash.
+
+- When changing any fetcher parameters, always update the output hash.
+  Use one of the methods from [](#sec-pkgs-fetchers-updating-source-hashes).
+  Otherwise, existing store objects that match the output hash will be re-used rather than fetching new content.
+
+A similar problem arises while testing changes to a fetcher's implementation.
+If the output of the derivation already exists in the Nix store, test failures can go undetected.
+The [`invalidateFetcherByDrvHash`](#tester-invalidateFetcherByDrvHash) function helps prevent reusing cached derivations.
```

Comment:
```suggestion
  Otherwise, existing store objects that match the output hash will be re-used rather than fetching new content.

  :::{.note}
  A similar problem arises while testing changes to a fetcher's implementation.
  If the output of the derivation already exists in the Nix store, test failures can go undetected.
  The [`invalidateFetcherByDrvHash`](#tester-invalidateFetcherByDrvHash) function helps prevent reusing cached derivations.
  :::
```